server/src/main/java/org/elasticsearch/action/search/TransportSearchHelper.java

```diff
@@ -108,6 +109,10 @@ static ParsedScrollId parseScrollId(String scrollId) {
         }
     }
 
+    static boolean hasLocalIndices(String scrollId) {
+        return Arrays.stream(parseScrollId(scrollId).getContext()).anyMatch(c -> c.getClusterAlias() == null);
```

Comment:
I would suggest that `hasLocalIndices` be a method of `ParsedScrollId` , it's more OO this way.

Reply:
You are absolutely right. Now I wonder why I didn't do it initially ...